### ISSUE TITLE
Hide annotate cells should toggle

### ISSUE THREAD

**Issue by sofroniewn** [hci-unihd/covid-if-annotations#24]
Hide annotate cells should probably be a toggle that goes "hide" -> "show" annotated cells and changes it's label based on button state, as it was a little unclear to me how it functioned. If you are in "hide mode" I'd probably at most show the last edited cell (so you can keep interacting with it), but then as soon as you label a new one that last one disappears. I imagine that auto disappearing will be very satisfying!

**Comment by sofroniewn** [comment]
I feel like I'd also have "hide" make the points, invisible instead of just the cell outlines. Will help reduce clutter. You can do this by hijaking the alpha value of the face and edge color, which can be done with something like

```python
viewer.layers['infected-vs-control'].edge_color[indices_of_annotated_cell, -1] = 0
viewer.layers['infected-vs-control'].face_color[indices_of_annotated_cell, -1] = 0
viewer.layers['infected-vs-control].refresh()
```

If you want them to appear again you can just set it to `1`. Note you need the refresh here to tigger the update as you are editing the array in place

**Comment by sofroniewn** [comment]
Let me know if you run into problems with this, I'm sure I could help out

**Comment by constantinpape** [comment]
> Hide annotate cells should probably be a toggle that goes "hide" -> "show" annotated cells and changes it's label based on button state, as it was a little unclear to me how it functioned. If you are in "hide mode" I'd probably at most show the last edited cell (so you can keep interacting with it), but then as soon as you label a new one that last one disappears. I imagine that auto disappearing will be very satisfying!

Yes, I agree. That would be even better than the current implementation and I also thought about this already.
I am not quite sure how to implement this through the `QButton` though.
Any pointers would be very welcome ;).

**Comment by constantinpape** [comment]
> I feel like I'd also have "hide" make the points, invisible instead of just the cell outlines. Will help reduce clutter.

I agree and I tried to implement this using your suggestion:
https://github.com/hci-unihd/covid-if-annotations/blob/master/napari_covid_if_annotations/_key_bindings.py#L103-L112

However, this doesn't quite work as expected, it only hides the edges, not the faces, see screenshots of normal / "hidden" mode:
![Screenshot from 2020-05-31 12-43-27](https://user-images.githubusercontent.com/4263537/83350469-9fa55800-a33c-11ea-8911-d264ae98f544.png)
![Screenshot from 2020-05-31 12-43-32](https://user-images.githubusercontent.com/4263537/83350470-a207b200-a33c-11ea-89ad-4394c6c4038a.png)

**Comment by sofroniewn** [comment]
Ah yes, my mistake, this is because the face color is being set through the `face_color_cycle` property. You should be able to do

```python
viewer.layers['infected-vs-control'].edge_color[indices_of_annotated_cell, -1] = 0
viewer.layers['infected-vs-control'].face_color_cycle[1:, -1] = 0
viewer.layers['infected-vs-control'].refresh_colors()
```

Where here you are making the "non white" colors invisible. You could also use an `edge_color_cycle` which had 4 colors for your 4 states - starting all black, but then toggling those edited 3 to invisible. It would bring a bit more symmetry to the code above, but up to you!

I will look into the `if layer.visible:` guard you put in to fix #23, I think we should be able to figure out another solution

**Comment by sofroniewn** [comment]
I can make a PR with this fix and also swap the `QPushButton` to a `QCheckBox` right now
